src/pycram/ros_utils/force_torque_sensor.py

```diff
@@ -77,3 +82,152 @@ def _stop_publishing(self) -> None:
         """
         self.kill_event.set()
         self.thread.join()
+
+
+class ForceTorqueSensor:
+    """
+    Monitor a force-torque sensor of a supported robot and save relevant data.
+
+    Apply a specified filter and save this data as well.
+    Default filter is the low pass filter 'Butterworth'
+
+    Can also calculate the derivative of (un-)filtered data
+    """
+    filtered = 'filtered'
+    unfiltered = 'unfiltered'
+
+    def __init__(self, robot_name, filter_config=FilterConfig.butterworth, filter_order=4, custom_topic=None,
+                 debug=False):
+
+        """
+        Create a subscriber for the force-torque-sensor topic of a specified robot.
+
+        :param robot_name: Name of the robot
+        :param filter_config: Desired filter (default: Butterworth)
+        :param filter_order: Order of the filter. Declares the number of elements that delay the sampling
+        :param custom_topic: Declare a custom topic if the default topics do not fit
+        """
+
+        self.robot_name = robot_name
+        self.filter_config = filter_config
+        self.filter = self.__get_filter(order=filter_order)
+        self.debug = debug
+
+        self.wrench_topic_name = custom_topic
+        self.force_torque_subscriber = None
+        self.init_data = True
+
+        self.whole_data = None
+        self.prev_values = None
+
+        self.order = filter_order
+
+        self.__setup()
+
+    def __setup(self):
```

Comment:
double underscore methods usually have a meaning of "super private please do never access this method from any other file". Is this intended?